docs/content/develop/properties.md

```diff
@@ -0,0 +1,107 @@
++++
+title = "Define properties"
+description = ""
+weight = 45
+draft = false
+bref = ""
+toc = true
+menu = "develop"
+tags = [ "develop" ]
++++
+
+Property definitions provide a standard, streamlined way to maintain state within your component, receive external input (via attributes), initialize values, trigger handler functions, and more.
+
+## Introduction
+
+To introduce the topic, consider a "Counter" component that counts how many times something has happened.  Such a component could define a counter like this.
+
+```javascript
+static get properties() {
+  return {
+    count: {
+      type: Number,
+      default: 0
+    }
+  };
+}
+```
+
+Defining `count` in this way brightens your day in the following ways:
+
+ - **A property**: The property's name is `count`, so `this.count` is created and initialized to the default value, the number `0`.  
+ - **An attribute**: An attribute, `pfe-c-count` is also created, and will be kept in sync with `this.count`.  The `pfe-c-` prefix is explained [below](#prefix).
+
+Below, all the options for property definitions are explained.
+
+## Property / attribute binding
+
+As mentioned in the introduction, defining a property also creates an attribute, and the two values will be kept in sync.  For properties with the String type, this is straightforward.  Attribute values are always strings, and so logic is needed to 
+
+### Handling `null`
+
+If a property is assigned to null, the associated attribute will be removed.
+
+Example: `this.count = null` will result in `this.removeAttribute("pfe-c-count")`.
+
+## Type
+
+Three property types are supported: String, Number, and Boolean.
+
+### String
+
+```
+name: {
+  type: String
+}
+```
+
+### Boolean
+
+### Number
+
```

Comment:
```suggestion
Similar to boolean, though attribute values are strings, if you are assigning a numerical value to your attribute, please use this type for your property.  Consider the use-case of `tabindex` for example.  This will output a quoted string when assigned to the attribute but use a numerical type when you query for the property.
```

Is this an int or does this support floats?

Reply:
It uses JS's built-in Number type, which is a floating point type, but floats can represent integers.  That's mentioned, but kind of buried, in the note about precision.  I feel like the name "Number" implies that any kind of number is supported (within the precision limits), but I'm fine with making it explicit if you think it would help.

Reply:
It uses JS's built-in Number type, which is a floating point type, but floats can represent integers.  That's mentioned, but kind of buried, in the note about precision.  I feel like the name "Number" implies that any kind of number is supported (within the precision limits), but I'm fine with making it explicit if you think it would help.

About the paragraph encouraging contributors to use `type: Number` for storing numbers, I feel like that would fit perfectly in a guide.  I see this doc as something like API docs, describing all the things you can do and how they work, whereas describing what you _should_ do would fit better in a guide, such as "Creating your first component" or "Adding typed properties to your component".  Does that make sense?  What do you think?

Reply:
Since we don't really have a guide yet outside of these docs, maybe we can insert it here for now and pull it out into the guide when we're building it?